### ISSUE TITLE
Meta-analysis: Adjusting Forest Plot

### ISSUE THREAD

**Issue by brunoverschuere** [jasp-stats/jasp-issues#967]
Problem: Its currently not possible to adjust the forest plot in the meta-analysis module

Request: It would be helpful if you could order the studies in the forest plot along effect size (from low>high, or vice versa) 

thank you!
Bruno

**Comment by boutinb** [comment]
@sophieberkhout, can you answer this request?

**Comment by sophieberkhout** [comment]
Hi Bruno, 

Is this about the classical meta-analysis? 
The Bayesian meta-analysis, which I am responsible for, does offer this functionality!

Thanks,
Sophie

**Comment by brunoverschuere** [comment]
Hi Sophie

Yes, for the classic meta

I checked, and indeed, for the Bayesian meta-analysis you can order on ascending/descending/row order, so it would be helpful if the same feature would be available for the classic meta-analysis

Thank you
Bruno
Ps as an interim solution, I realized I can reorder the excel file, but evidently it would be cool to have the feature within JASP

**Comment by boutinb** [comment]
@sophieberkhout Is it something you could do? Because I'm not sure that the creator of the classical Meta Analysis will do it.

**Comment by brunoverschuere** [comment]
@sophieberkhout as the average effect for the Bayesian isnt exactly the same as that of the Classic meta, I cannot use the forest plot generated by the Bayesian when reporting results of the classic meta. You think the ordering feature implemented in the Bayesian can also be implemented for the classic meta ?

**Comment by sophieberkhout** [comment]
Sorry for the late reply. It should be possible to add this to the classical meta-analysis also.

@boutinb I will have a look at it, as it seems like it should not be that difficult to implement.

**Comment by raoelg** [comment]
@brunoverschuere Hi Bruno, I think it is in general very difficult to anticipate all possible ordering needs in a meta-analysis. That's why the order in the forest plot etc follows the order in the data so that it is under full control of the user. Of course this requires you to make copies with different orders, or you double click the data and do it in Excel. It's far more useful if there will be an ordering feature in the data view than in every analysis module separately.
So @boutinb is that going to be implemented in the future? That you can order your data rows from within the data view? I'm sure that will also resolve such issues in other modules would be a future module proof add on.

**Comment by boutinb** [comment]
Hi @raoelg You can reorder the levels/labels of a column when you click on its header in the data view. But if I do that I don't see that the order of the forest plot changes. That's weird because if I check for example in the Descriptive analysis, the Frequency table, then the row order follows the order of the labels.
Maybe @vandenman can give a hint why this order is not taken into account in Classical Meta-Analysis ?
Also if there are some really common ordering like effect size ordering, it can be a good idea to add an option for this (for example a dropdown beside the 'Forest plot' checkbox).

**Comment by raoelg** [comment]
I think that only changes the labeling order in a factor (i.e., the mapping
between a label and the underlying numerical label that is used in models
is changed; for instance the mapping A=1, B=2, C=3, D=4 can be changed to
B=1, A=2, D=3, C=4), that has consequences in the parameterization of
statistical models, but that's not the issue here. The order of the rows in
the table should be changeable. The forest plot (and presumably many other
modules) simply used the order of the data in the table.

I don't think adding complexity to the meta-analysis is going to be useful,
because the choice for which ordering to allow for is going to be
necessarily limited and arbitrary. There will always be someone who wants
to order it on a different basis. It's better if the rows in the data file
can be reordered physically from within the data view so that you don't
have to go outside of jasp to rearrange the rows.



On Thu, Nov 26, 2020 at 10:02 AM bruno boutin <notifications@github.com>
wrote:

> Hi @raoelg
> <https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fraoelg&data=04%7C01%7CR.P.P.P.Grasman%40uva.nl%7C8d4aeb3d508849aa747408d891e9f505%7Ca0f1cacd618c4403b94576fb3d6874e5%7C1%7C0%7C637419781307325589%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=7QhRi9P%2Fr7%2FVbfj7Eq1NseeZLYCWzC%2FD437S0jgvrTs%3D&reserved=0>
> You can reorder the levels/labels of a column when you click on its header
> in the data view. But if I do that I don't see that the order of the forest
> plot changes. That's weird because if I check for example in the
> Descriptive analysis, the Frequency table, then the row order follows the
> order of the labels.
> Maybe @vandenman
> <https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fvandenman&data=04%7C01%7CR.P.P.P.Grasman%40uva.nl%7C8d4aeb3d508849aa747408d891e9f505%7Ca0f1cacd618c4403b94576fb3d6874e5%7C1%7C0%7C637419781307335543%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=HtKQtZ906BoLTL9WaFty2jo36RPxF6h0lHAm4l9c9Zg%3D&reserved=0>
> can give a hint why this order is not taken into account in Classical
> Meta-Analysis ?
> Also if there are some really common ordering like effect size ordering,
> it can be a good idea to add an option for this (for example a dropdown
> beside the 'Forest plot' checkbox).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjasp-stats%2Fjasp-issues%2Fissues%2F967%23issuecomment-734168008&data=04%7C01%7CR.P.P.P.Grasman%40uva.nl%7C8d4aeb3d508849aa747408d891e9f505%7Ca0f1cacd618c4403b94576fb3d6874e5%7C1%7C0%7C637419781307335543%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=piq1G%2B012VtSFbF9wIevonC7lnHaZ5ywPIViuV2c0tg%3D&reserved=0>,
> or unsubscribe
> <https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAB2SOSP6FV2OJWZGNJRQ7MDSRYKRBANCNFSM4Q6JMJFA&data=04%7C01%7CR.P.P.P.Grasman%40uva.nl%7C8d4aeb3d508849aa747408d891e9f505%7Ca0f1cacd618c4403b94576fb3d6874e5%7C1%7C0%7C637419781307335543%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=dJ%2FBn7esnyxlz6hIpnGMGEKFMJVzOyzPH4pXnrLtKDc%3D&reserved=0>
> .
>


-- 
Department of Psychological Methods,
University of Amsterdam,
Nieuwe Prinsengracht 130
1018VZ Amsterdam
The Netherlands

tel: +31 (20) 525 6738
http://raoul.socsci.uva.nl

**Comment by boutinb** [comment]
Hi @raoelg, maybe I was not clear. I did not say that values of the labels should be changed, it is the order of these labels that can be changed.
Here below, I have used the arrows at the right side to set the 'Aronson' label at the second position.
<img width="604" alt="Screenshot 2020-11-27 at 09 59 40" src="https://user-images.githubusercontent.com/503332/100430603-4faf1000-3097-11eb-90fc-43d34d64b195.png">
